app/models/history.rb

```diff
@@ -33,6 +33,10 @@ class History < ApplicationRecord
       where('histories.created_at >= ?', 24.hours.ago)
     when 'Last 14 Days'
       where('histories.created_at >= ? AND histories.created_at < ?', 14.days.ago.to_date.to_datetime, Date.today.to_datetime)
+    when 'Total'
```

Comment:
the same scoping function is implemented in other models as well and I think it might be good to keep them consistent:
https://github.com/SaraAlert/SaraAlert/blob/master/app/models/transfer.rb#L29
https://github.com/SaraAlert/SaraAlert/blob/master/app/models/patient.rb#L237

I was also thinking it would be good to change the data type of `time_frame` to a symbol (like `:last_24_hours`, `:last_14_days`, `:total`) to avoid potentially bugs in the future due to typos. That doesn't have to be done now, but if you think that's a good idea and want to do that, the changes just need to be made here:

[https://github.com/SaraAlert/SaraAlert/blob/master/app/jobs/cache_analytics_job.rb#L46](https://github.com/SaraAlert/SaraAlert/blob/master/app/jobs/cache_analytics_job.rb#L46)

and here:
[https://github.com/SaraAlert/SaraAlert/blob/master/test/jobs/analytics_job_test.rb#L279](https://github.com/SaraAlert/SaraAlert/blob/master/test/jobs/analytics_job_test.rb#L279)

Reply:
Yes I will make a change for the two other spots you linked